src/Transport/Connectivity/NamespaceManagerAdapter.cs

```diff
@@ -19,6 +19,22 @@ public NamespaceManagerAdapter(NamespaceManager manager)
 
         public Uri Address => _manager.Address;
 
+        public bool HasManageRights
+        {
+            get
+            {
+                try
+                {
+                    _manager.GetQueues();
+                    return true;
+                }
+                catch
```

Comment:
We need to be careful here in order not to confuse `UnauthorizedAccessException` for anything else.